### DISCUSSION TITLE
9-slicing Sprites

### DISCUSSION THREAD

**Discussion by leonidcan** [AvaloniaUI/Avalonia#12842] [Ideas]
Can we have 9-slicing Sprites in AvaloniaUI?

**Comment by timunie** [comment]
what is that? do you have a link for us?

**Reply by BreyerW** [reply]
This technique is most common in web and in 3d texturing mostly of in-game or in-tool UI elements. It refers to virtually slicing texture in a way that corners are NOT scaled, left and right slices are scaled only vertically, bottom and top only horizontally and middle in both directions. Eg imagine button texture with borders on that texture if u put it as-is the borders would scale horribly but with 9-slicing they wouldnt become pixelated. Theres even wikipedia entry on it under 9-slice scaling [https://en.m.wikipedia.org/wiki/9-slice_scaling](https://en.m.wikipedia.org/wiki/9-slice_scaling)

**Reply by kekekeks** [reply]
@timunie https://learn.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.image.ninegrid?view=winrt-22621

**Reply by timunie** [reply]
is that something we want to have in Avalonia.Labs?

**Comment by Sorien** [comment]
way more interesting is 9-patch variant, its good candidate for 3rd party control

**Comment by Shadowblitz16** [comment]
This is necessary for any sort of pixel art theme.
It's not a control it's a way to draw an image.

**Comment by kekekeks** [comment]
For it to work properly we'll need a way to support DPI-specific assets. Otherwise there will be scaling artifacts.

**Reply by cyraid** [reply]
@kekekeks it would be nice to have. I've come across wanting to have a pixel art style theme for my Editor, and putting zoomed pixel art images will be fine as long as it uses nearest neighbour. Like `RenderOptions.BitmapInterpolationMode="None"`
